### ISSUE TITLE
Why not normalize the output from BERT in ContrastiveEmbed? 

### ISSUE THREAD

**Issue by TsingWei** [IDEA-Research/GroundingDINO#150]
In ContrastiveEmbed, the simple dot production between queries and encoded text features is used as 'similarity', or, the classification logits.  In the query-selection stage, the queries from the encoder are LayerNormed, while they are not normalized in the decode stage. 

Just wondering, why the dot product of two vectors can be used as the 'contrastive' logits, without any normalization?

**Comment by xiexie123** [comment]
actually, In ContrastiveEmbed, the text features is out from Feature Enhancer with also a LayerNorm rather than the original embedding out from BERT. Technically, In ContrastiveEmbed both queries and encoded text features is 'Norm'.
